autograder/cli/common.py

```diff
@@ -8,33 +17,100 @@
+        expected_user_info_type = COURSE_USER_INFO_TYPE
+        if (len(keys) == 0):
+            keys = COURSE_USER_HEADERS
     else:
-        _list_users(users)
+        expected_user_info_type = SERVER_USER_INFO_TYPE
+        if (len(keys) == 0):
+            keys = SERVER_USER_HEADERS
 
-def _list_users(users, indent = ''):
-    users = list(users)
+    rows = []
+    for user in users:
+        if (user['type'] != expected_user_info_type):
+            raise ValueError("Invalid user type for listing users table: Expected: '%s',"
+                    + " actual: '%s'.", expected_user_info_type, user['type'])
+
+        row = [user[key] for key in keys]
+        rows.append(row)
+
+    _print_tsv(rows, header, keys)
+
+def _list_course_users(users, indent = ''):
+    for user in users:
+        if (user['type'] != COURSE_USER_INFO_TYPE):
+            raise ValueError("Invalid user type for listing course users: Expected: '%s',"
+                    + " actual: '%s'.", COURSE_USER_INFO_TYPE, user['type'])
 
+        print(indent + "Email: " + user['email'])
+        print(indent + "Name: " + user['name'])
+        print(indent + "Role: " + user['role'])
+        print(indent + "LMS ID: " + user['lms-id'])
+
+def _list_server_users(users, indent = ''):
     for i in range(len(users)):
         user = users[i]
 
+        if (user['type'] != SERVER_USER_INFO_TYPE):
+            raise ValueError("Invalid user type for listing server users: Expected: '%s',"
+                    + "actual: '%s'.", SERVER_USER_INFO_TYPE, user['type'])
+
         if (i != 0):
             print()
 
-        print(indent + "Email:", user['email'])
-        print(indent + "Name:", user['name'])
-        print(indent + "Role:", user['role'])
-        print(indent + "LMS ID:", user['lms-id'])
+        print(indent + "Email: " + user['email'])
+        print(indent + "Name: " + user['name'])
+        print(indent + "Role: " + user['role'])
+        print(indent + "Courses:")
 
-def _list_users_table(users, header = True, keys = HEADERS):
-    if (header):
-        print("\t".join(keys))
+        for j, course in enumerate(user['courses']):
+            if (j != 0):
+                print()
 
+            print(indent + INDENT + "ID: " + user['courses'][course]['id'])
+            print(indent + INDENT + "Name: " + user['courses'][course]['name'])
+            print(indent + INDENT + "Role: " + user['courses'][course]['role'])
+
+def _list_server_users_table_normalize(users, header = True, keys = BASE_USER_HEADERS):
+    rows = []
     for user in users:
+        if (user['type'] != SERVER_USER_INFO_TYPE):
+            raise ValueError("Invalid user type for listing normalized server users table:"
+                    + " Expected: '%s', actual: '%s'.", SERVER_USER_INFO_TYPE, user['type'])
+
         row = [user[key] for key in keys]
-        print("\t".join([str(value) for value in row]))
+        if ((user.get('courses') is None) or (len(user['courses']) == 0)):
```

Comment:
This comment requires no action.

For cases like this, I often prefer:
```
    if (len(user.get('courses', [])) == 0):
```

Note that your version also protects against null/nil/None values.